apis/r/R/datasets.R

```diff
@@ -0,0 +1,87 @@
+#' SOMA Example Datasets
+#'
+#' @description
+#' Access example SOMA objects bundled with the tiledbsoma package.
+#'
+#' Use `list_datasets()` to list the available datasets and `load_dataset()` to
+#' load a dataset into memory using the appropriate SOMA class. The
+#'  `extract_dataset()` method returns the path to the extracted dataset without
+#'  loading it into memory.
+#'
+#' @details
+#' The SOMA objects are stored as `tar.gz` files in the package's `extdata`
+#' directory. Calling `load_dataset()` extracts the `tar.gz` file to the
+#' specified `dir`, inspects its metadata to determine the appropriate SOMA
+#' class to instantiate, and returns the SOMA object.
+#'
+#' @examples
+#' soma_pbmc_small <- load_dataset("soma-exp-pbmc-small")
+#'
+#' @name example-datasets
+NULL
+
+#' @rdname example-datasets
+#' @return
+#'  - `list_datasets()` returns a character vector of the available datasets.
+#' @importFrom tools file_path_sans_ext
+#' @export
+list_datasets <- function() {
+  data_dir <- example_data_dir()
+  stopifnot("The dataset directory does not exist" = dir.exists(data_dir))
+  files <- dir(data_dir, pattern = "tar.gz")
```

Comment:
Not that it'd ever happen, but `pattern = "tar.gz"` will match all files matching `tar[ANY_SINGLE_CHARACTER]gz`; if you wanted to be more precise in matching, you could adjust this to `pattern = "tar\\.gz$`

```R
> # we only want to match x.tar.gz since it's the only tarball
> list.files('~/scratch')
[1] "tar.gz.x" "tarXgz"   "txt"      "x.tar.gz"
> list.files('~/scratch', pattern = 'tar.gz')
[1] "tar.gz.x" "tarXgz"   "x.tar.gz"
> list.files('~/scratch', pattern = 'tar\\.gz$')
[1] "x.tar.gz"
```

Reply:
Good call.